src/main/java/ch/njol/skript/util/EnumUtils.java

```diff
@@ -43,55 +40,67 @@ public EnumUtils(final Class<E> c, final String languageNode) {
 		
 		this.c = c;
 		this.languageNode = languageNode;
-		
+
 		names = new String[c.getEnumConstants().length];
 		
-		Language.addListener(new LanguageChangeListener() {
-			@Override
-			public void onLanguageChange() {
-				validate(true);
-			}
-		});
+		Language.addListener(() -> validate(true));
 	}
 	
 	/**
 	 * Updates the names if the language has changed or the enum was modified (using reflection).
 	 */
-	final void validate(final boolean force) {
+	void validate(final boolean force) {
 		boolean update = force;
-		
-		final int newL = c.getEnumConstants().length;
-		if (newL > names.length) {
-			names = new String[newL];
+
+		int newLength = c.getEnumConstants().length;
+		if (newLength != names.length) {
+			names = new String[newLength];
 			update = true;
 		}
-		
+		for (E constant : c.getEnumConstants()) {
+			if (!parseMap.containsValue(constant)) { // A new value was added to the enum
+				update = true;
+				break;
+			}
+		}
+
 		if (update) {
 			parseMap.clear();
 			for (final E e : c.getEnumConstants()) {
-				final String[] ls = Language.getList(languageNode + "." + e.name());
-				names[e.ordinal()] = ls[0];
-				for (final String l : ls)
-					parseMap.put(l.toLowerCase(), e);
+				String key = languageNode + "." + e.name();
+				int ordinal = e.ordinal();
+
+				for (String option : Language.getList(key)) {
+					option = option.toLowerCase();
+					NonNullPair<String, Integer> strippedOption = Noun.stripGender(option, key);
+
+					if (names[ordinal] == null) { // Add to name array if needed
+						names[ordinal] = strippedOption.getFirst();
```

Comment:
Again, ideally we'd get a variable for this since we're using it 3+ times, but it's not a problem.